3-enrich/scala-common-enrich/src/main/scala/com.snowplowanalytics.snowplow.enrich/common/enrichments/registry/CurrencyConversionEnrichment.scala

```diff
@@ -0,0 +1,155 @@
+/*
+ * Copyright (c) 2012-2015 Snowplow Analytics Ltd. All rights reserved.
+ *
+ * This program is licensed to you under the Apache License Version 2.0,
+ * and you may not use this file except in compliance with the Apache License Version 2.0.
+ * You may obtain a copy of the Apache License Version 2.0 at http://www.apache.org/licenses/LICENSE-2.0.
+ *
+ * Unless required by applicable law or agreed to in writing,
+ * software distributed under the Apache License Version 2.0 is distributed on an
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the Apache License Version 2.0 for the specific language governing permissions and limitations there under.
+*/
+package com.snowplowanalytics
+package snowplow
+package enrich
+package common
+package enrichments
+package registry
+
+// Java
+import java.net.UnknownHostException
+
+// Maven Artifact
+import org.apache.maven.artifact.versioning.DefaultArtifactVersion
+
+// Scalaz
+import scalaz._
+import Scalaz._
+
+// json4s
+import org.json4s.JValue
+
+// Iglu
+import iglu.client.{
+  SchemaKey,
+  SchemaCriterion
+}
+import iglu.client.validation.ProcessingMessageMethods._
+
+// Joda-Time
+import org.joda.time.DateTime
+
+// Scala-Forex
+import com.snowplowanalytics.forex.oerclient.{
+  OerClientConfig,
+  AccountType,
+  DeveloperAccount,
+  EnterpriseAccount,
+  UnlimitedAccount,
+  OerResponseError
+}
+import com.snowplowanalytics.forex.{
+  Forex,
+  ForexConfig
+}
+
+// This project
+import common.utils.ConversionUtils
+import utils.ScalazJson4sUtils
+
+/**
+ * Companion object. Lets us create an CurrencyConversionEnrichment
+ * instance from a JValue.
+ */
+object CurrencyConversionEnrichmentConfig extends ParseableEnrichment {
+
+  val supportedSchema = SchemaCriterion("com.snowplowanalytics.snowplow", "currency_conversion_config", "jsonschema", 1, 0)
+
+  // Creates a CurrencyConversionEnrichment instance from a JValue
+  def parse(config: JValue, schemaKey: SchemaKey): ValidatedNelMessage[CurrencyConversionEnrichment] = {
+    isParseable(config, schemaKey).flatMap( conf => {
+      (for {
+        apiKey        <- ScalazJson4sUtils.extract[String](config, "parameters", "apiKey")
+        baseCurrency  <- ScalazJson4sUtils.extract[String](config, "parameters", "baseCurrency")
+        accountType   <- (ScalazJson4sUtils.extract[String](config, "parameters", "accountType") match {
+          case Success("DEVELOPER")  => DeveloperAccount.success
+          case Success("ENTERPRISE") => EnterpriseAccount.success
+          case Success("UNLIMITED")  => UnlimitedAccount.success
+
+          // These failures should be prevented by the schema
```

Comment:
"should" is a bit ambiguous in a comment - people can mistake it for "a developer really ought to implement this in the future".

So better would be:

// Should never happen (prevented by schema validation)